src/include/function/list/operations/list_extract_operation.h

```diff
@@ -26,10 +26,12 @@ struct ListExtract {
     template<typename T>
     static inline void operation(
         ku_list_t& list, int64_t pos, T& result, ValueVector& resultValueVector) {
-        auto uint64Pos = (uint64_t)pos;
-        if (list.size < uint64Pos) {
-            throw RuntimeException("list_extract(list, index): index=" + TypeUtils::toString(pos) +
-                                   " is out of range.");
+        if (pos == 0 || abs(pos) > list.size) {
+            return;
+        }
+        auto uint64Pos = (pos < 0) ? (pos + list.size + 1) : (uint64_t)pos;
+        if (uint64Pos == 0) {
+            uint64Pos = list.size;
```

Comment:
I don't see changes applied to string list extraction. Make sure their behaviour is consistent and add test.

Reply:
string extract cases are handled in the array extract function, only 1 case was missing which was the -ve index out of range, I've added logic and test for it